File: tasks/implementations/php/tasks/api_client.php
<?php
function run_api_client(string $size, string $fixturesRoot): void {
  $text = file_get_contents($fixturesRoot . "/mocks/public_api_{$size}.json");
  preg_match_all('/"id":\s*(\d+),\s*"name":\s*"([^"]+)",\s*"value":\s*(\d+)/', $text, $matches, PREG_SET_ORDER);
  $total = 0;
  foreach ($matches as $match) $total += intval($match[1]) + intval($match[3]) + strlen($match[2]);
  echo $total . PHP_EOL;
}

File: tasks/implementations/go/tasks/api_client.go
package main

import (
    "fmt"
    "os"
    "path/filepath"
    "regexp"
    "strconv"
)

func RunApiClient(size string, fixturesRoot string) {
    data, _ := os.ReadFile(filepath.Join(fixturesRoot, "mocks", "public_api_"+size+".json"))
    re := regexp.MustCompile(`"id":\s*(\d+),\s*"name":\s*"([^"]+)",\s*"value":\s*(\d+)`)
    total := 0
    for _, match := range re.FindAllStringSubmatch(string(data), -1) {
        id, _ := strconv.Atoi(match[1])
        value, _ := strconv.Atoi(match[3])
        total += id + value + len(match[2])
    }
    fmt.Println(total)
}
